src/query_plan/operation.rs

```diff
@@ -869,6 +891,34 @@ pub(crate) mod normalized_field_selection {
             &mut self.data.sibling_typename
         }
 
+        pub(crate) fn type_if_added_to(
+            &self,
+            parent_type: &CompositeTypeDefinitionPosition,
+            schema: &ValidFederationSchema,
+        ) -> Option<CompositeTypeDefinitionPosition> {
+            let data = self.data();
+            if &data.field_position == parent_type {
+                let digest = match &self.data.field_position {
+                    FieldDefinitionPosition::Object(ty) => CompositeTypeDefinitionPosition::Object(
+                        ObjectTypeDefinitionPosition::new(ty.type_name.clone()),
+                    ),
+                    FieldDefinitionPosition::Interface(ty) => {
+                        CompositeTypeDefinitionPosition::Interface(
+                            InterfaceTypeDefinitionPosition::new(ty.type_name.clone()),
+                        )
+                    }
+                    FieldDefinitionPosition::Union(ty) => CompositeTypeDefinitionPosition::Union(
+                        UnionTypeDefinitionPosition::new(ty.type_name.clone()),
+                    ),
+                };
+                return Some(digest);
+            }
+            if data.name() == parent_type.type_name() {
```

Comment:
this actually should be checking whether field name is [`__typename`](https://github.com/apollographql/federation-next/blob/main/src/query_plan/operation.rs#L51)

Reply:
I have the if condition changed to check for `__typename`. Do you also want to the `return` call changed or is it ok as is?